reliquery/metadata.py

```diff
@@ -495,4 +504,105 @@ def get_by_relic_tag(self, tag: RelicTag) -> List:
         except Error as e:
             logging.warning(f"Error getting relic tag: {e}")
 
-        return list(map(RelicTag.parse_sql_result, tags))
+        return [RelicTag.parse_sql_result(i, tag.relic) for i in tags]
+
+    def sync_relic_data(self, relic_data: RelicData) -> RelicData:
+        relic = self.get_relic_data_by_name(
+            relic_data.relic_name, relic_data.relic_type, relic_data.storage_name
+        )
+
+        if relic:
+            return relic
+        else:
+            self._create_relic_data(
+                relic_data.relic_name, relic_data.relic_type, relic_data.storage_name
+            )
+            return self.get_relic_data_by_name(
+                relic_data.relic_name, relic_data.relic_type, relic_data.storage_name
+            )
+
+    def get_relic_data_by_name(
+        self, relic_name: str, relic_type: str, storage_name: str
+    ) -> RelicData:
+
+        try:
+            cur = self.conn.cursor()
+            cur.execute(
+                """
+                SELECT * FROM relics
+                WHERE relic_name = ?
+                AND relic_type = ?
+                AND storage_name =?
+                LIMIT 1
+                """,
+                (relic_name, relic_type, storage_name),
+            )
+            rows = cur.fetchall()
+            if rows:
+                return RelicData.parse_dict(RelicData.parse_sql_result(rows[0]))
+            else:
+                return None
+
+        except Error as e:
+            logging.warning(
+                "Error getting Relic data by name: "
+                + f"{relic_name, relic_type, storage_name} | {e.__class__()}: {e}"
+            )
+
+    def _create_relic_data(
+        self, relic_name: str, relic_type: str, storage_name: str
+    ) -> None:
+        try:
+            cur = self.conn.cursor()
+            cur.execute(
+                """
+            INSERT INTO relics VALUES(?,?,?,?,?)
+            """,
+                (
+                    None,
+                    relic_name,
+                    relic_type,
+                    storage_name,
+                    dt.datetime.utcnow().strftime(dt_format),
+                ),
+            )
+
+            self.conn.commit()
+
+        except Error as e:
+            logging.warning(f"Error creating Relic data: {e}")
```

Comment:
not sure we should catch these things. If thhere is an actual erorr and we just let it slide, now the system is in a confusing state and we may get weird results. probably better to just let the error happen